crates/prover/src/runner.rs

```diff
@@ -231,6 +234,18 @@ where
         let skip_submission_until_l1 = std::env::var("SKIP_PROOF_SUBMISSION_UNTIL_L1")
             .map_or(0u64, |v| v.parse().unwrap_or(0));
 
+        if let Some(config) = &self.pruning_config {
+            let pruner = Pruner::<DB>::new(
+                config.clone(),
+                self.ledger_db.get_last_pruned_l2_height()?.unwrap_or(0),
+                self.soft_confirmation_tx.subscribe(),
+                self.ledger_db.clone(),
+            );
+
+            self.task_manager
+                .spawn(|cancellation_token| pruner.run(cancellation_token));
+        }
+
```

Comment:
Should we prune prover?

Reply:
For the prover, i don't think we need to keep all the history in order for us to be able to prove sequencer commitments.
For the sequencer, same reasoning applies, as long as we don't use it as an archive node (no full or prover nodes are syncing from there). So basically we can have multiple sequencers running (maybe one or two of which are archive and the rest are pruned).
That is, if we have a high-availability setup

Reply:
talked f2f we might turn off if necessary